### ISSUE TITLE
Discontinue support of pre-KitKat devices?

### ISSUE THREAD

**Issue by moezbhatti** [moezbhatti/qksms#548]
Users with an Android version before KitKat account for approximately 7% of QKSMS' active users

By removing support for these devices, they'll still have access to all releases prior to the change, but only users with KitKat+ will be able to update.

Removing it will allow us to make the codebase substantially cleaner, as KitKat introduced a ton of official APIs for SMS and MMS. We'll be able to get rid of a lot of code and probably fix a few bugs in the process.

I'm leaning towards stripping out support in favor of a cleaner codebase and less legacy code. Anyone have any input on this?

**Comment by searchy2** [comment]
A lot of developers have stopped supporting pre KitKat. I believe that going forward, supporting only KitKat and above versions is the way to go. However, there should be a plan for the transition and I believe that dropping support should coincide with the new features you have planned and a version bump to 3.0. 
But before pre-KitKat devices are dropped from continued development, I think QKSMS needs a few additional features in the attachments category. Ability to add voice notes, contacts, location, and appointments would make QKSMS competitive with default solutions. 
Basically, build QKSMS to have all standard SMS features for legacy OS. Then add new features for v3.0.

**Comment by tomose** [comment]
Agreed, pre KK support should be removed.

**Comment by M2ck** [comment]
Yes you should discontinue support for pre KK devices.

**Comment by moezbhatti** [comment]
Thanks for the feedback
I like your suggestion @searchy2, this would allow a major refactor of the entire SMS/MMS codebase, and warrant a version 3.0
